src/inverted-index.js

```diff
@@ -72,31 +80,79 @@ class InvertedIndex {
       });
       // Sort the index keys
       index = InvertedIndex.sortObjectKeys(index);
-      // Simply update the this.indexes object
-      // Don't return it yet
+      /** Simply update the this.indexes object
+       *  Don't return it yet
+       */
       this.indexes[filename] = index;
     }
   }
 
   /**
+   * Returns all indexes or an index for a specified file
+   *
+   * @param{String} filename - Optional parameter specifying a file whose index is required
    * @returns{Object} - Object that contains all the index
    */
-  getIndex() {
-    return this.indexes;
+  getIndex(filename) {
+    return (filename === undefined) ? this.indexes : this.indexes[filename];
   }
 
   /**
+   * Function to iterate over the indexes object and returns the search results
+   *
+   * @param{Array} words - An array of words to be searched
+   * @param{Object} indexes - An object containing inverted indexes
+   * @param{String} fileName - Optional argument indicating the filename
+   * @returns{Object} results - An object containing the search results
+   */
+  static iterCollection(words, indexes, fileName) {
+    // Object to store search results
+    const results = {};
+    // check if fileName is undefined
+    for (const file in indexes) {
```

Comment:
### Critical

@andela-rbabalola The logic in this entire loop needs to be ironed out. If a file name is passed, there is no need to loop through the entire collection of indices, however in your implementation, you are looping through the entire thing and on line `123`,  while still within the `loop of the entire files' indices`, you are then accessing the index for a specific file.

That should have been done outside of a loop.

The logic to search is simple, if file is passed, get index for the particular file, for each word, find the documents. If no file is passed, run through the indices for each file. The logic for looping through a collection of indices and searching/matching for each search term should be abstracted and be called separately for each of the 2 scenarios given earlier